src/state/thunks/fetchMappedSet.ts

```diff
@@ -39,7 +39,7 @@ export function fetchMappedSet(visualId: VisualId, mappingId: string, date: Date
     const timeKey = formatUTCDate(date);
     const loadingKey = `loading-${mappingId}-${timeKey}`;
 
-    dispatch(AppApi.pushLoading(loadingKey));
+    dispatch(() => AppApi.pushLoading(loadingKey));
```

Comment:
`AppApi.pushLoading` returns a functions, which will be called by dispatch. This way the wrapper method gets called, never calling the function returned by pushLoading, actually doing nothing. So the state is not changed at all, but also we never get a loading bar.

Reply:
well all of this seems correct. except there is a loading bar. :>

Reply:
oh got it. so thunks basically do this already. the loading bar gets activated by mapgl. for some reason i initially didn't see `AppApi.pushLoading` dispatched anywhere else, which caused my confusion. 

is there any other way to prevent the components from updating while rendering? except explicitly set `isLoading` until the initial render is complete, i guess, which seems like a hack.